pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/FormActivityDao.java

```diff
@@ -52,24 +57,72 @@
     @CreateSqlObject
     TemplateDao getTemplateDao();
 
+    @CreateSqlObject
+    ActivitySql getActivitySql();
+
+
+    // Convenience helper for inserting activity without any nested activities.
+    default void insertActivity(FormActivityDef activity, long revisionId) {
+        insertActivity(activity, List.of(), revisionId);
+    }
+
     /**
-     * Create new form activity by inserting all related data for activity definition. Use this to create the first version of an activity.
+     * Create new top-level activity by inserting all related data for activity definition. Use this to create the first
+     * version of an activity. Any child activities that the activity references (such as in a "nested activity block")
+     * need to be passed in the list of nested activity definitions.
      *
      * <p>See {@link ActivityDao#insertActivity(FormActivityDef, RevisionMetadata)} for a more convenient way for
-     * inserting, and {@link ConsentActivityDao#insertActivity(ConsentActivityDef, long)} specifically for consent activities. Changing
-     * content should be done with the appropriate add/disable methods.
+     * inserting, and {@link ConsentActivityDao#insertActivity(ConsentActivityDef, long)} specifically for consent
+     * activities. Changing content should be done with the appropriate add/disable methods.
      *
-     * @param activity   the activity to create, without generated things like ids
-     * @param revisionId the revision to use, will be shared for all created data
+     * @param activity         the activity to create, without generated things like ids
+     * @param nestedActivities the list of nested activities to created, that the activity references.
+     * @param revisionId       the revision to use, will be shared for all created data
      */
-    default void insertActivity(FormActivityDef activity, long revisionId) {
+    default void insertActivity(FormActivityDef activity, List<FormActivityDef> nestedActivities, long revisionId) {
```

Comment:
I have the nested activities as a separate list here because if I'm understanding correctly, we want parent/child to be more-or-less standalone activities, and just have them "reference" each other. E.g. child references parent with just a `parentActivityCode` property.

However, I've pretty much made it a requirement that nested activities are created alongside the parent (so you have to pass the child activities along in this list). This is because I think we have a "chicken-and-egg" problem. Parent and child activities reference each other, so we need to create them within the same "transaction".